azure-pipelines.yml

```diff
@@ -100,7 +100,7 @@ jobs:
       - script: 'pip install -U numpy~=1.21.0'
         displayName: 'Upgrade numpy'
         
-      - script: 'pip install pytest pytest-runner jupyter jupyter-client nbconvert nbformat seaborn xgboost tqdm && pip list && python setup.py pytest'
+      - script: 'pip install "pytest<7.2.0" pytest-runner jupyter jupyter-client nbconvert nbformat seaborn xgboost tqdm && pip list && python setup.py pytest'
```

Comment:
Consider leaving a comment indicating what the problem is with pytest 7.2 so that we can remove this limitation when appropriate.

Reply:
Actually I was able to remove the version cap by additionally installing py. 

Relevant changelog which mentions pytest removed its dependency on py. https://docs.pytest.org/en/7.2.x/changelog.html#pytest-7-2-0-2022-10-23

Notably our errors with pytest 7.2.0 were related to py
<img width="430" alt="image" src="https://user-images.githubusercontent.com/22120253/198376793-14b94332-022d-4141-933d-4dfb33814554.png">